### PR TITLE
support Kimi-K2.5-w4a8 on ascend

### PR DESCRIPTION
**Pull request by khalil2ji3mp6**
## Motivation

We have adapted **ModelSlim quantization** for the **Kimi-K2.5-W4A8** model on **Ascend**.
 Model weights: https://modelscope.cn/models/Eco-Tech/Kimi-K2.5-W4A8

## Modifications

1. As **ModelSlim** needs to identify the quantization type of weights, we add corresponding fields to the **prefix** (e.g., `language.` and `vision_tower.encoder.blocks.`).
2. For the **ModelSlim** case, the ViT linear layers are replaced with `ReplicatedLinear` to create `weight_scale` and `weight_offset`.

## Accuracy Tests

```sh
python -m sglang.launch_server --skip-server-warmup  \
    --model-path /xxx/Kimi-K2.5-w4a8 --quantization modelslim --dtype bfloat16 \
    --host 0.0.0.0 --port 8100 \
    --trust-remote-code --device npu --attention-backend ascend \
    --tp-size 16 --base-gpu-id 0 --mem-fraction-static 0.82 --max-running-requests 72 \
    --disable-radix-cache --chunked-prefill-size 16384 --context-length 8192 --max-prefill-tokens 16384 \
    --enable-multimodal --mm-attention-backend ascend_attn --sampling-backend ascend \
    --enable-dp-attention --dp-size 2 --enable-dp-lm-head \
    --moe-a2a-backend deepep --deepep-mode auto \
    --cuda-graph-bs 32 40
```

gsm8k:

```sh
python3 -m sglang.test.few_shot_gsm8k \
    --num-questions 200 \
    --num-shots 5 \
    --data-path /xxx/gsm8k.jsonl \
    --max-new-tokens 512 \
    --parallel 72 \
    --host http://0.0.0.0 \
    --port 8100
```

image

image:

```sh
curl --location 'http://0.0.0.0:8100/generate' --header 'Content-Type: application/json' --data '{
    "text": "Describe this image:\n<|media_pad|>",
    "image_data": ["/xxx/1080p.jpg"],
    "stream": false,
    "sampling_params": {
        "temperature": 0,
        "max_new_tokens": 128
    }
}'
```

```json
{"text":"This is a stunning landscape photograph capturing a serene alpine lake surrounded by lush green mountains under a bright blue sky.\n\n**The Lake:** The central focus is a large, calm body of water with an extraordinary turquoise-blue color. The surface is so still that it acts like a perfect mirror, reflecting the sky, clouds, and surrounding mountains with remarkable clarity.\n\n**The Mountains:** Towering, verdant mountains rise on both sides of the lake. They are covered in dense forests of deep green trees, with patches of lighter green grassy slopes. The peaks are partially obscured by soft, white clouds, adding depth and atmosphere. On the right side","output_ids":[2512,387,261,18742,18252,13499,48926,261,100417,98973,24247,28703,673,53515,8372,26107,1638,261,13293,9447,18985,1481,381,1008,13830,16397,646,12383,4962,387,261,4393,11,26389,4120,318,3431,472,519,30821,95183,45362,3278,13,646,9612,387,971,3095,473,483,21928,1517,261,4734,22612,11,54335,276,18985,11,41365,11,316,16699,26107,472,30503,37366,1481,381,1008,46753,16397,37092,288,11,102737,632,26107,16023,418,2957,15595,318,276,24247,13,3350,554,11724,306,34474,44511,318,8055,8372,13900,11,472,42898,318,35985,8372,145654,63938,13,646,54877,554,36668,50058,4247,673,9875,11,6032,41365,11,11100,12466,316,19397,13,3282,276,2029,4594],"meta_info":{"id":"550dbd782d184e4d8681263d1fdce8e4","finish_reason":{"type":"length","length":128},"prompt_tokens":2695,"weight_version":"default","total_retractions":0,"completion_tokens":128,"cached_tokens":0,"cached_tokens_details":null,"dp_rank":0,"e2e_latency":10.388509310025256,"response_sent_to_client_ts":1773026922.343935}}
```

gpu Kimi-K2.5 unquant test

```sh
python3 -m sglang.launch_server  \
    --model-path /xxx/Kimi_K2_5 \
    --host 0.0.0.0 --port 30012  \
    --trust-remote-code  \
    --mem-fraction-static 0.9 \
    --tp-size 8
```

image

```json
{"text":"This is a stunning landscape photograph capturing a serene alpine lake surrounded by lush green mountains. The lake's water is a striking turquoise color, so clear and calm that it perfectly mirrors the sky, clouds, and surrounding peaks like a glass surface.\n\nIn the foreground, leafy green branches frame the scene on both the left and right sides, adding depth and a natural border to the composition. The left side features darker green foliage, while the right side shows leaves with hints of autumn yellow.\n\nThe middle ground is dominated by the expansive lake, which stretches across the image. On the far shore, rolling hills covered in vibrant green grass and dense forests rise up. A small cluster of traditional alpine houses with dark roofs is nestled into the hillside on the right, adding a charming human element to the otherwise wild landscape.\n\nThe background features majestic mountains with forested slopes that climb toward rocky peaks. The sky above is a bright blue with wispy white clouds scattered across it, suggesting a pleasant, sunny day. The overall","output_ids":[2512,387,261,18742,18252,13499,48926,261,100417,98973,24247,28703,673,53515,8372,26107,13,646,24247,1048,3431,387,261,34962,95183,3278,11,971,4522,316,26389,473,483,16762,46988,276,18985,11,41365,11,316,16699,54877,1517,261,9452,9612,1481,767,276,61275,11,116144,8372,25759,6999,276,12265,418,2957,276,3550,316,2029,15595,11,11100,12466,316,261,6623,5341,308,276,25165,13,646,3550,4594,5062,53797,8372,75023,11,2396,276,2029,4594,7431,14190,472,41378,318,45262,17294,1481,1008,9947,7519,387,49641,673,276,72557,24247,11,1155,68497,5080,276,4082,13,3282,276,4972,37816,11,26812,34896,11724,306,30282,8372,20579,316,34474,44511,16023,979,13,401,3307,12216,318,9205,98973,17104,472,9292,72640,387,101157,1576,276,144836,418,276,2029,11,11100,261,38266,6325,5545,308,276,9000,11145,18252,1481,1008,4631,5062,83025,26107,472,17304,295,63938,473,21743,14177,72553,54877,13,646,18985,4681,387,261,13293,9447,472,284,124814,6032,41365,57521,5080,483,11,44244,261,25986,11,41149,2422,13,646,10390],"meta_info":{"id":"1a01f00ad4d04e60868aebf3fd255a62","finish_reason":{"type":"length","length":200},"prompt_tokens":2695,"weight_version":"default","total_retractions":0,"completion_tokens":200,"cached_tokens":0,"cached_tokens_details":null,"e2e_latency":6.418799877166748,"response_sent_to_client_ts":1773049617.8235707}}
```

## Checklist

- [ ] Format your code according to the [[Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit)](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [[Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests)](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [[Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations)](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [[Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy)](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [[Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed)](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [[guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance)](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [[PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process)](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [[CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS)](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [[comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests)](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/models/kimi_vl_moonvit.py [hunk 1]
```diff
@@ -393,21 +396,50 @@ class MLP2(nn.Module):
...
+                bias=bias,
+                quant_config=quant_config,
+                prefix=add_prefix("fc1", prefix),
+            )
+        else:
+            self.fc0 = nn.Linear(dims[0], dims[1], bias=bias)
+            self.fc1 = nn.Linear(dims[1], dims[2], bias=bias)
+
         self.activation = activation
-        for m in [self.fc0, self.fc1]:
```
Comment (xiaobaicxy) [inline python/sglang/srt/models/kimi_vl_moonvit.py:402]: Do not remove this codes

Comment (khalil2ji3mp6) [reply python/sglang/srt/models/kimi_vl_moonvit.py:402]: Added it back. Thanks for pointing it out.

**Review by iforgetmyname — APPROVED** [review]
